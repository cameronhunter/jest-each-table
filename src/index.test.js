const createTestTable = require('./index');

const testcases = createTestTable(
  [...Array(10)].map((_, i) => ({
    inputs: { left: i, right: i },
    output: i + i
  }))
);

const testsuite = test.each(...testcases);

testsuite('Testcase $testcaseIndex: $inputs.left + $inputs.right = $output', ({ inputs, output }) => {
  expect(inputs.left + inputs.right).toBe(output);
});

test('can be used to create strings', () => {
  const [quasis, ...expressions] = testcases;
  expect(String.raw({ raw: quasis }, ...expressions.map((expression) => JSON.stringify(expression)))).toMatchSnapshot();
});
