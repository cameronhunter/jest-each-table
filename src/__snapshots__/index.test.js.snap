// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be used to create strings 1`] = `
"
testcaseIndex|inputs|output
1|{\\"left\\":0,\\"right\\":0}|0
2|{\\"left\\":1,\\"right\\":1}|2
3|{\\"left\\":2,\\"right\\":2}|4
4|{\\"left\\":3,\\"right\\":3}|6
5|{\\"left\\":4,\\"right\\":4}|8
6|{\\"left\\":5,\\"right\\":5}|10
7|{\\"left\\":6,\\"right\\":6}|12
8|{\\"left\\":7,\\"right\\":7}|14
9|{\\"left\\":8,\\"right\\":8}|16
10|{\\"left\\":9,\\"right\\":9}|18
"
`;
